crates/serialize_hierarchy/src/lib.rs

```diff
@@ -262,6 +262,144 @@ impl<T> SerializeHierarchy for Vec<T> {
     }
 }
 
+impl<T: Serialize + DeserializeOwned> SerializeHierarchy for Vector2<T> {
+    fn serialize_path<S>(&self, path: &str) -> Result<S::Serialized, Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let idx = ["x", "y"].into_iter().position(|x| x == path);
+        match idx {
+            Some(i) => S::serialize(&self[i]).map_err(Error::SerializationFailed),
+            _ => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn deserialize_path<S>(
+        &mut self,
+        path: &str,
+        _data: S::Serialized,
+    ) -> Result<(), Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let idx = ["x", "y"].into_iter().position(|x| x == path);
+        match idx {
+            Some(i) => {
+                let deserialized = S::deserialize(_data).map_err(Error::DeserializationFailed)?;
+                self[i] = deserialized;
+                Ok(())
+            }
+            None => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
+    }
+
+    fn exists(_path: &str) -> bool {
+        matches!(_path, "x" | "y")
+    }
```

Comment:
Remove the underscore:

```suggestion
    fn exists(path: &str) -> bool {
        matches!(path, "x" | "y")
    }
```